README.md

```diff
@@ -1019,6 +1019,255 @@ Voilà! Since we updated `index.html`, you'll need to restart the dev server. Go
+
+We are going to be needing these two functions, both can be placed outside the
+`make` function of `App.re`.
+
+The first one takes the old feed and a new one, and proceeds to merge both:
+
+```reason
+let mergeFeeds = (oldFeed: Feed.feed, newFeed: Feed.feed): Feed.feed => {
+  let combinedEntries = Array.concat([oldFeed.entries, newFeed.entries]);
+  {entries: combinedEntries};
+};
+```
+
+The second one is just taking the feed render code and moving it to a separate
+function so we can reuse it:
+
+```reason
+let renderFeed = (feed: Feed.feed) =>
+  <div>
+    <h1> {React.string("GitHub Feed")} </h1>
+    <ul>
+      {switch (feed.entries) {
+       | [||] => React.string("This user feed is empty")
+       | entries =>
+         entries
+         |> Array.map((entry: Feed.entry) =>
+              <li key={entry.id}>
+                <h2> {React.string(entry.title)} </h2>
+                {switch (entry.content) {
+                 | None => React.null
+                 | Some(content) =>
+                   <p dangerouslySetInnerHTML={"__html": content} />
+                 }}
+              </li>
+            )
+         |> React.array
+       }}
+    </ul>
+  </div>;
+```
+
+#### 11.4: Modify `fetchFeed` to support pagination
+
+Update `fetchFeed` to accept a `page` parameter and include it in the API
+request. We’ll use [labeled
+arguments](https://reasonml.github.io/docs/en/function#named-arguments) to avoid
+confusion when calling this function:
+
+```reason
+let fetchFeed = (~username, ~page) => {
+  setState(state =>
+    {
+      ...state,
+      step:
+        switch (state.step) {
+        | Loaded(r)
+        | Loading(Some(r)) => Loading(Some(r))
+        | Loading(None)
+        | Idle
+        | InvalidUsername => Loading(None)
+        },
+    }
+  );
+  module P = Js.Promise;
+  Fetch.fetch(
+    "https://gh-feed.vercel.app/api?user="
+    ++ Username.toString(username)
+    ++ "&page="
+    ++ string_of_int(page),
+  )
+  |> P.then_(response => {
+        let status = Fetch.Response.status(response);
+        if (status === 200) {
+          /* If status is OK, proceed to parse the response */
+          response
+          |> Fetch.Response.text
+          |> P.then_(text =>
+              {
+                let data =
+                  try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                  | Json.Decode.DecodeError(msg) =>
+                    Js.Console.error(msg);
+                    Error("Failed to decode: " ++ msg);
+                  };
+
+                switch (data) {
+                | Error(_) =>
+                  setState(state => {...state, step: Loaded(data)})
+                | Ok(data) =>
+                  setState(state => {
+                    let updatedFeed =
+                      switch (state.step) {
+                      | Loaded(Ok(feed))
+                      | Loading(Some(Ok(feed))) =>
+                        mergeFeeds(feed, data)
+                      | _ => data
```

Comment:
```suggestion
                      | Loading(Some(Ok(feed))) =>
                        mergeFeeds(feed, data)
                      | Loaded(_) => data // unreachable
                      | _ => data
```

Similar issue as the comment above, at this point it shouldn't be in `Loaded(_)` state

Reply:
See [above](https://github.com/ahrefs/react-alicante-workshop/pull/9#discussion_r1732822938).